sdk/communication/communication-chat/src/chatClient.ts

```diff
@@ -341,12 +341,32 @@ export class ChatClient {
    */
   public on(event: "participantsRemoved", listener: (e: ParticipantsRemovedEvent) => void): void;
 
-  public on(event: ChatEventId, listener: (e: any) => void): void {
+  /**
+   * Subscribe function for realTimeNotificationConnected.
+   * @param event - The realTimeNotificationConnected Event
+   * @param listener - The listener to handle the event.
+   */
+  public on(event: "realTimeNotificationConnected", listener: () => void): void;
+
+  /**
+   * Subscribe function for realTimeNotificationDisconnected.
+   * @param event - The realTimeNotificationDisconnected Event
+   * @param listener - The listener to handle the event.
+   */
+  public on(event: "realTimeNotificationDisconnected", listener: () => void): void;
+
+  public on(
+    event: ChatEventId | "realTimeNotificationConnected" | "realTimeNotificationDisconnected",
+    listener: (e?: any) => void
+  ): void {
     if (this.signalingClient === undefined) {
       throw new Error("Realtime notifications are only supported in the browser.");
     }
-
-    if (!this.isRealtimeNotificationsStarted) {
+    if (
+      !this.isRealtimeNotificationsStarted &&
+      event !== "realTimeNotificationConnected" &&
+      event !== "realTimeNotificationDisconnected"
```

Comment:
it feels weird that I can't subscribe to these before calling `startRealtimeNotifications` - I would imagine listening to connect/disconnect *before* connecting makes perfect sense.

Reply:
These two additional checks are actually here so that you can subscribe connect/disconnect without calling startRealtimeNotifications first. Previously if you haven't called startRealtimeNotifications, you won't be able to subscribe to events. In another word, the error will be thrown only when real-time notification hasn't started AND this subscription is not regarding trouter connect/disconnect.

Reply:
Ahh now I see!